references/cli/adopt.go

```diff
@@ -160,6 +160,9 @@ func (opt *AdoptOptions) Complete(f velacmd.Factory, cmd *cobra.Command, args []
 		}) {
 			opt.AppName = opt.NativeResourceRefs[0].Name
 		}
+		if opt.AppNamespace == "" {
+			opt.AppNamespace = opt.NativeResourceRefs[0].Namespace
```

Comment:
1. what if the silce is zero?
2. what if the namespace is still empty? Can we downgrade to the environment( from vela env and the kubeconfig context)?

Reply:
1. from previous code, if opt.parseResourceRef didn't return err, the slice would not be zero, else it would not reach the code
2. i will reconsider about the downgrade solution

Reply:
velacmd.GetNamespace is already able to downgrade to env and kubeconfig context when -n is not set, but after testing, i find the env usage is: 
- vela adopt deployment/kube-system/metrics-server -e default   (default namespace is set)
- vela adopt deployment/kube-system/metrics-server -e namespace=default (default namespace is not set, but this is more human-readable)

Reply:
velacmd.GetNamespace function is already able to downgrade to env and kubeconfig context when -n is not set, but i think the env logic is too subtle, it would match the first namespace with label `namespace.oam.dev/env` except default namespace, here is the case:

- default namespace is set
```
vela adopt deployment/monitoring-system/metrics-server -e default
```
- kube-system namespace is not set
```
vela adopt deployment/monitoring-system/metrics-server -e kube-system
```
- vela-system namespace is set as it has label `namespace.oam.dev/env=syc-vela-system`
```
vela adopt deployment/monitoring-system/metrics-server -e syc-vela-system
```

Reply:
> velacmd.GetNamespace function

Great, just use this function should be fine.

> the env logic is too subtle

What's the results for these cases? I think we're discusing the case that we didn't specify any env or namespace, but your case specify env, then just use the env logic should be fine?

Reply:
The env logic would match the first namespace with label `namespace.oam.dev/env`, so the downgrade is  --env flags, then kubeconfig context.
I am not sure about the expect "env" downgrade is --env flags or just environment variable like `KUBE_NAMESPACE=kube-system`

Reply:
As the velacmd.GetNamespace is called many times, in this issue i think the env logic shoue be fine

Reply:
if velacmd.GetNamespace works, then the `opt.AppNamespace` won't be empty, why could your changes take effect?

As to this issue, I think we need this case work:

```
vela adopt deployment/metrics-server --apply
```

It will adopt deployment in the default namespace named `metrics-server` to be an application in the default namespace.

Reply:
Coud you add a test case for that?

Reply:
In velacmd.GetNamespace, the namespace comes from `--namespace > --env > kubeconfig context`, so if we don't provide any args above, the namespace would be empty.

My changes use the namespace of the k8s object resourcesRef (deployment/kube-system/metrics-server) when velacmd.GetNamespace get empty namespace, it would be the last downgrade choice, so this case can work:

```
$ vela adopt deployment/kube-system/metrics-server --apply
resources adopted in app kube-system/metrics-server
```

If we need this case work, it needs to change the opt.parseResourceRef and i will make another commit
```
vela adopt deployment/metrics-server --apply
```

Reply:
```
vela adopt deployment/metrics-server --apply
```

I think it means using the "default" namespace for deployment metrics-server.

Reply:
OK, the last commit implement this case:

```
$vela adopt deployment/nginx-deployment --apply
resources adopted in app default/nginx-deployment
```

Reply:
great, can you help add some test cases

Reply:
done, add two test cases without -n

```
$ vela adopt deployment/kube-system/metrics-server --apply
resources adopted in app kube-system/metrics-server

$vela adopt deployment/nginx-deployment --apply
resources adopted in app default/nginx-deployment
```